### PR TITLE
Moved initialization logic from Operator to Propagator

### PR DESCRIPTION
**Pull request by vincepandolfo**
Some initialization logic for the Propagator was part of the Operator code. 

This PR moves it down to the Propagator.

### CODE REVIEW

File: devito/operator.py [hunk 1]
```diff
@@ -165,12 +165,14 @@ def __init__(self, nt, shape, dtype=np.float32, stencils=[],
         self.stencils = [Eq(expr_indexify(eqn.lhs), expr_indexify(eqn.rhs))
                          for eqn in self.stencils]
 
+        for name in factorized.keys():
+            self.propagator.factorized[name] = expr_indexify(factorized[name])
```
Comment (FabioLuporini) [inline devito/operator.py]: is this identical to : 
```
for name, value in factorized.items():
    self.propagator.factorized[name] = expr_indexify(value)
```
?

Comment (vincepandolfo) [inline devito/operator.py]: Yes. And it also doesn't make much sense to have it there. I'll change this

File: devito/propagator.py [hunk 2]
```diff
@@ -35,10 +37,11 @@ class Propagator(object):
                       a list stating block sizes for each dimension. Set block size to None to skip blocking on that dim
     """
 
-    def __init__(self, name, nt, shape, spc_border=0, time_order=0,
+    def __init__(self, name, nt, shape, stencils, factorized=[], spc_border=0, time_order=0,
```
Comment (FabioLuporini) [inline devito/propagator.py]: Never use ``[]`` for initialisation. Use ``None`` and then in the constructor you can do

```
self.factorized = factorized or []
```

Explanation here: http://docs.python-guide.org/en/latest/writing/gotchas/

**Comment by FabioLuporini** [comment]
Apart from these minor things, this should be merged ASAP.

Also, since you're touching this file, can you break the doc strings longer than 89 characters ? Thanks

**Comment by mlange05** [comment]
Looks good to me.

**Comment by FabioLuporini** [comment]
Merging this.

**Pull request merged**